internal/coreinternal/processor/filterspan/filterspan.go

```diff
@@ -124,12 +128,21 @@ func (mp *propertiesMatcher) MatchSpan(span pdata.Span, resource pdata.Resource,
 	return mp.PropertiesMatcher.Match(span.Attributes(), resource, library)
 }
 
-// serviceNameForResource gets the service name for a specified Resource.
-func serviceNameForResource(resource pdata.Resource) string {
+// serviceNameForResource gets the service name for a specified Resource
+func serviceNameForResource(resource pdata.Resource) (string, bool) {
 	service, found := resource.Attributes().Get(conventions.AttributeServiceName)
 	if !found {
-		return "<nil-service-name>"
+		return "<nil-service-name>", found
+	}
+	return service.AsString(), found
+}
+
+// serviceNameForSpan gets the service name for a span
+func serviceNameForSpan(span pdata.Span) (string, bool) {
+	service, found := span.Attributes().Get(conventions.AttributeServiceName)
```

Comment:
`service.name` is a [Resource semantic convention](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#service). It is not supposed to be recorded as a Span attribute. Why is this check?